service/src/main/java/bio/terra/profile/app/configuration/AzureConfiguration.java

```diff
@@ -62,11 +62,27 @@ public void setAuthorizedUserKey(String authorizedUserKey) {
     }
   }
 
-  public TokenCredential buildManagedAppCredentials() {
-    return new ClientSecretCredentialBuilder()
-        .clientId(managedAppClientId)
-        .clientSecret(managedAppClientSecret)
-        .tenantId(managedAppTenantId)
+  public ChainedTokenCredential createChainedCredential() {
```

Comment:
This should return `TokenCredential` and be called `createTokenCredential`, the fact that it is chained is an implementation detail